### PR TITLE
doc: update building guide

### LINKED ISSUE #562: Not able to build Notation project from source
### What is the areas you experience the issue in?

Notation CLI

### What is not working as expected?

After following the [instructions](https://github.com/Manas23601/notation/blob/main/building.md) to install notary from source. I'm unable to view the new changes while running `docker --help`. 

### What did you expect to happen?

I'm failing to see the following in `docker --help`

```
Management Commands:
  generate*   Generate artifacts (CNCF Notary Project, 0.1.0)
  notation*   Manage signatures on Docker images (CNCF Notary Project, 0.5.3-alpha)

```

### How can we reproduce it?

Follow the instructions to build from source

### Describe your environment

Installation method:Git
OS:  Ubuntu 22.04
Shell: bash
Golang Version : 20.1

### What is the version of your Notation CLI or Notation Library?

```
Notation: Notary v2, A tool to sign, store, and verify artifacts.

Version:     v1.0.0-rc.1+unreleased
Go version:  go1.20
Git commit:  f463f0cb16753d5e398b3bb2d6e26a0e23e519c7
```

go.mod file
```
module github.com/notaryproject/notation

go 1.19

require (
	github.com/docker/docker-credential-helpers v0.7.0
	github.com/notaryproject/notation-core-go v1.0.0-rc.1
	github.com/notaryproject/notation-go v1.0.0-rc.1.0.20230203031935-510def1a3f48
	github.com/opencontainers/go-digest v1.0.0
	github.com/opencontainers/image-spec v1.1.0-rc2
	github.com/sirupsen/logrus v1.9.0
	github.com/spf13/cobra v1.6.1
	github.com/spf13/pflag v1.0.5
	oras.land/oras-go/v2 v2.0.0
)

require (
	github.com/Azure/go-ntlmssp v0.0.0-20221128193559-754e69321358 // indirect
	github.com/fxamacker/cbor/v2 v2.4.0 // indirect
	github.com/go-asn1-ber/asn1-ber v1.5.4 // indirect
	github.com/go-ldap/ldap/v3 v3.4.4 // indirect
	github.com/golang-jwt/jwt/v4 v4.4.3 // indirect
	github.com/inconshreveable/mousetrap v1.1.0 // indirect
	github.com/veraison/go-cose v1.0.0 // indirect
	github.com/x448/float16 v0.8.4 // indirect
	golang.org/x/crypto v0.5.0 // indirect
	golang.org/x/mod v0.7.0 // indirect
	golang.org/x/sync v0.1.0 // indirect
	golang.org/x/sys v0.4.0 // indirect
)

```

### PR DESCRIPTION
**Pull request by JeyJeyGao**
Resolves #562 
Signed-off-by: Junjie Gao <junjiegao@microsoft.com>

### CODE REVIEW

File: building.md [hunk 1]
```diff
@@ -17,13 +17,14 @@ Building above binaries require [golang](https://golang.org/dl/) with version `>
...
-  # output
+  # expected output
   /home/<user>/bin/notation
   ```
+
+  If you confront `notation not found`, please add `~/bin/` to your $PATH:
+  ```sh
+  export PATH="$HOME/bin:$PATH"
+  ```
+  If you would like to add the path permanently, please add it to your `~/.profile` file.
```
Comment (patrickzheng200) [inline building.md:29]: we could provide more details here, such as what content should be added into the `~/.profile` file?

Comment (JeyJeyGao) [reply building.md]: Updated.

File: building.md [hunk 2]
```diff
@@ -17,13 +17,18 @@ Building above binaries require [golang](https://golang.org/dl/) with version `>
...
   ```
+
+  If you confront `notation not found`, please add `~/bin/` to your $PATH:
+  ```sh
+  export PATH="$HOME/bin:$PATH"
+  ```
+  If you would like to add the path permanently, add the command to your `~/.profile` file:
+  ```sh
+  echo 'export PATH="$HOME/bin:$PATH"' >> ~/.profile
+  source ~/.profile
```
Comment (priteshbandi) [inline building.md:32]: Depending upon user's shell the location of profile will change. For Eg: 
* bash: `~/.bash_profile` or `~/..profile`
* zsh: `~/.zprofile` 
* ksh: `~/..profile`

Comment (JeyJeyGao) [reply building.md:32]: Updated.

**Review by patrickzheng200 — DISMISSED** [review]
LGTM

**Review by FeynmanZhou — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

**Review by FeynmanZhou — APPROVED** [review]
LGTM

File: building.md [hunk 3]
```diff
@@ -17,13 +17,22 @@ Building above binaries require [golang](https://golang.org/dl/) with version `>
...
+  ```
+  If you would like to add the path permanently, add the command to your Bash `~/.profile` file:
+  ```sh
+  echo 'export PATH="$HOME/bin:$PATH"' >> ~/.profile
+  source ~/.profile
+  ```
+  For other shells, please add the command to the corresponding `profile` file.
+  - Bash: ~/.bash_profile or ~/.profile
+  - Zsh: ~/.zprofile
+  - Ksh: ~/.profile
```
Comment (shizhMSFT) [inline building.md:37]: nit:
```suggestion
  - Bash: `~/.bash_profile` or `~/.profile`
  - Zsh: `~/.zprofile`
  - Ksh: `~/.profile`
```

File: building.md [hunk 4]
```diff
@@ -17,13 +17,22 @@ Building above binaries require [golang](https://golang.org/dl/) with version `>
...
   /home/<user>/bin/notation
   ```
+
+  If you confront `notation not found`, please add `~/bin/` to your $PATH:
+  ```sh
+  export PATH="$HOME/bin:$PATH"
+  ```
+  If you would like to add the path permanently, add the command to your Bash `~/.profile` file:
+  ```sh
+  echo 'export PATH="$HOME/bin:$PATH"' >> ~/.profile
```
[the remark this answers is not in the document]
Comment (JeyJeyGao) [reply building.md]: Thanks Shiwei! Update the doc to let the user decide the default shell profile.

File: building.md (hunk 3 above)
Comment (JeyJeyGao) [reply building.md]: Updated.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by patrickzheng200 — APPROVED** [review]
LGTM

**Review by FeynmanZhou — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

**Pull request merged**